src/filesystem.cpp

```diff
@@ -111,10 +217,41 @@ void FileSystemAccess::unescapefsincompatible(string* name) const
     }
 }
 
+std::string FileSystemAccess::getPathSeparator(const char *path)
```

Comment:
Hi, I think this one should be platform specific - eg. on unix a \ is allowed in a file's name, and is not a separator.  thanks

Reply:
This idea behind this method is look for the first occurrence of \ or / so:
- If you are in Windows the paths you will provide are like this: C:\\ (First separator found is \)
- If you are in Unix/APPLE the paths you will provide are like this: /home/folder (First separator found is /)

Reply:
Anyway as you said this method should be platform specific, so I think this code makes it easier and more robust.
```
std::string FileSystemAccess::getPathSeparator(const char *path)
{
#if defined (__linux__) || defined (__ANDROID__) || defined  (__APPLE__)
return "/";
#elif defined(_WIN32) || defined(_WIN64)
return "\\";
#elif 
// Default case    
return "\\/";
#endif
}
```
What do you think?

Reply:
yes that looks fine to me, nice one!  thanks